### DISCUSSION TITLE
Testing Guards in Xstate with Angular

### DISCUSSION THREAD

**Discussion by emst92** [statelyai/xstate#3852] [Q&A]
I want to test the guards in the XState Machine

```
  {
    guards: {
      isSomething: context =>
        meetsConditionsForSomething(context),
      isSomethingElse: context =>
        meetsConditionsForSomethingElse(context)
    }
  }
```
I want to mock the context and test the Guards only. Is there a way to do that?

Is there a way to use machine.transition(...) or something similar?

To be clear I have no previous experience with testing in Angular applications.

**Comment by Andarist** [comment]
If you want to mock things out then you can always just test your `meetsConditionsForSomething` instead of `guards.isSomething`. If you want check if a given guard is being checked when something happens within the machine then you can use `machine.transition` for that:
```ts
machine.withContext(mockedContext).transition({ machine: 'value' }, { type: 'SOME_EVENT' })
```

**Comment by emst92** [comment]
Thanks for the quick answer! 

The guards in my case are more complexed than in the exemple I gave, there are more conditions under each guard.

I will try to accomplish my goal with you suggestion:
`machine.withContext(mockedContext).transition({ machine: 'value' }, { type: 'SOME_EVENT' })`
